src/praisonai-agents/praisonaiagents/tools/base.py

```
@@ -196,15 +203,28 @@ def safe_run(self, **kwargs) -> ToolResult:
             "function": {
                 "name": self.name,
                 "description": self.description,
                 "parameters": self.parameters
             }
         }
+        
+        # Apply dynamic override if present
+        if hasattr(self, '_schema_override') and self._schema_override is not None:
+            try:
+                return self._schema_override(base_schema)
+            except Exception as e:
```

Ruff (0.15.15)

[warning] Do not catch blind exception: `Exception`

(BLE001)